src/sources/opentelemetry/tests.rs

```diff
@@ -48,6 +48,7 @@ async fn receive_grpc_logs_vector_namespace() {
                 address: http_addr,
                 tls: Default::default(),
                 keepalive: Default::default(),
+                headers: Default::default(),
```

Comment:
I tried adding a test for the new functionality, but none of the existing tests exercise the OTLP/HTTP endpoints, so it was not clear how I could do that.

Reply:
I can see how this would add a lot of scope here. Since this change doesn't alter existing behavior, we can accept this contribution as is.

Reply:
I gave it another look and was able to figure out how to add the tests.